README.md

```diff
@@ -26,104 +26,142 @@ What is wrong with the classic format?
+[
+  [
+    author,
+    sequence,
+    previous,
+    timestamp,
+    [
+      content,
+      contentSignature
+    ]
+  ],
+  signature
+]
+```
 
-Message must conform to the following rules:
+Here is the same data as above, but displayed in raw binary (hexadecimal):
 
- - Must be in the format specified above
- - The `previous` field must be correct (reference the previous message)
- - The `signature` field must be valid
- - The [SSB-BFE] format for `author` and `previous` must stay the same
-(there is no upgradeability in the middle of a feed)
- - The `author` can not change between two messages (always stays the same)
- - The maximum size of a message in bytes must not exceed 8192 bytes
+```
+    Offset   00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F
+    000000   6C 6C 33 34 3A 00 03 5C 27 AC 6E F0 CD FB D0 F8   ll34:..\'¬nðÍûÐø
+    000010   9A 89 A1 B6 5A 36 04 77 A3 3E C7 9C B7 AB 14 CD   ..¡¶Z6.w£>Ç.·«.Í
+    000020   90 76 25 59 BE E2 FF 69 31 65 32 3A 06 02 69 31   .v%Y¾âÿi1e2:..i1
+    000030   32 33 34 35 65 6C 64 34 3A 74 65 78 74 31 35 3A   2345eld4:text15:
+    000040   06 00 47 6F 6F 64 20 6D 6F 72 6E 69 6E 67 21 34   ..Good morning!4
+    000050   3A 74 79 70 65 37 3A 06 00 67 72 65 65 74 65 36   :type7:..greete6
+    000060   36 3A 04 00 51 A6 7A 43 6A 66 F6 6D E0 3D 77 73   6:..Q¦zCjfömà=ws
+    000070   C0 B7 BA 98 84 61 32 46 C6 EE 6C 74 1B 1D 9E 59   À·º..a2FÆîlt...Y
+    000080   18 24 B3 C7 1D A3 EC 35 BF E0 32 CF 86 55 7C F8   .$³Ç.£ì5¿à2Ï.U|ø
+    000090   72 30 E9 56 8E D5 7B 25 F6 77 FE 58 3B 17 3D BD   r0éV.Õ{%öwþX;.=½
+    0000A0   E7 08 82 0F 65 65 36 36 3A 04 00 6D 57 9F 55 14   ç...ee66:..mW.U.
+    0000B0   D2 D8 69 09 AD 7B 31 F8 24 4F A7 FC 6A 0D C1 1E   ÒØi.­{1ø$O§üj.Á.
+    0000C0   F4 1A 92 71 86 FB 8D 1B FC D5 17 B3 88 05 F0 A6   ô..q.û..üÕ.³..ð¦
+    0000D0   48 AA BA 24 F4 46 B0 9E 65 64 B6 9A DE 97 F9 18   Hªº$ôF°.ed¶.Þ.ù.
+    0000E0   04 AF 5F 7A F3 5E 5D 4B FD 85 0B 65               .¯_zó^]Ký..e
+```
 
-Content must conform to the following rules:
 
- - a `type` field with a string value of only the following values
-   possible: `metafeed/add`, `metafeed/update` or `metafeed/tombstone`
- - a `subfeed` field with a [SSB-BFE] encoded feed id
- - a `metafeed` field with a [SSB-BFE] encoded feed id
- - a `nonce` field with 32 bytes of random data
- - the content signature must be correct
+## Validation
 
+For validation we differentiate between *message validity* and *content
+validity*. Since the content can be encrypted, there is potentially no way to
+validate that. This means a message should only be rejected before insertion
+into the local database if it fails the message validation rules. A message
+should not be included in the bendy butt feed if its content is not valid.
+
+The message **MUST** conform to the following rules:
+
+ - Has the shape of a bencode list `[payload, signature]
+ - The payload has the shape of bencode list `[author, sequence, previous, timestamp, contentSection]`
```

Comment:
I think this needs to go, as you can't check that if the content is encrypted.

Reply:
Or rather moved to content validation

Reply:
In the update I just made, `contentSection` is defined as *either* encrypted or _not_ encrypted, so we can keep this sentence as is, and it means that validation should check whether `contentSection` is a BFE box2 or a bencode list with two parts.